inference-engine/src/mkldnn_plugin/nodes/mkldnn_normalize_node.h

```diff
@@ -75,80 +78,95 @@ class MKLDNNNormalizeL2Node : public MKLDNNNode {
+    void getSupportedDescriptors() override {};
     void initSupportedPrimitiveDescriptors() override;
     void createPrimitive() override;
     bool created() const override;
     void execute(mkldnn::stream strm) override;
     bool canBeInPlace() const override {
         return false;
     }
-
-    static bool isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept;
     bool canFuse(const MKLDNNNodePtr& node) const override;
 
+    std::vector<VectorDims> shapeInfer() const override;
+    void prepareParams() override;
+    void executeDynamicImpl(mkldnn::stream strm) override { execute(strm); }
+
 private:
     enum class NormEpsMode {
         ADD,
         MAX
     };
-    NormEpsMode epsMode = NormEpsMode::ADD;
-
-    float epsApply(const float &modulo) const {
-        if (epsMode == NormEpsMode::ADD) {
-            return modulo + eps;
-        } else if (epsMode == NormEpsMode::MAX) {
-            return std::max(modulo, eps);
-        } else {
-            IE_THROW() << errorPrefix << "has unsupported epsilon mode";
-        }
-    }
-
-    bool cornerCase = false;
-
-    template<typename T>
-    struct NormalizeExecute;
-
-    template <typename in_data_t, typename out_data_t>
-    void normalize_nchw(const in_data_t* src_data, out_data_t* dst_data, const InferenceEngine::SizeVector& dims);
 
-    template <typename in_data_t, typename out_data_t>
-    void normalize_nchw_ref(const in_data_t* src_data, out_data_t* dst_data, const InferenceEngine::SizeVector& dims);
-
-    template <typename in_data_t, typename out_data_t>
-    void normalize_nhwc(const in_data_t* src_data, out_data_t* dst_data, const InferenceEngine::SizeVector& dims);
-
-    template <typename in_data_t, typename out_data_t>
-    void normalize_blk(const in_data_t* src_data, out_data_t* dst_data, const InferenceEngine::SizeVector& dims);
-
-    void setPostOps(mkldnn::primitive_attr &attr, bool initWeights = false);
-    inline void apply_post_ops_scalar(float &dst_value, int index_c);
-
-    template <typename in_data_t, typename out_data_t>
-    void normalize_function(const in_data_t* src_data, out_data_t* dst_data, const InferenceEngine::SizeVector& dims);
-
-    bool across_spatial = true;
-    float eps = 1e-10f;
-
-    InferenceEngine::Precision input_prec, output_prec;
-    size_t src_data_size, dst_data_size;
-
-    mkldnn::primitive_attr attr;
+    struct NormalizeL2Attrs {
+        NormEpsMode epsMode = NormEpsMode::ADD;
+        bool across_spatial = true;
+        bool cornerCase = false;
+        float eps = 1e-10f;
+
+        bool is_nchw = false;
+        bool is_nhwc = false;
+        bool is_blk = false;
+
+        InferenceEngine::Precision input_prec = Precision::UNSPECIFIED;
+        InferenceEngine::Precision output_prec = Precision::UNSPECIFIED;
+        size_t src_data_size = 0lu;
+        size_t dst_data_size = 0lu;
```

Comment:
minor: seems like it is enough to pass precisions only and use `in/output_prec.size()` inside the kernel.